tests/utils.bash

```diff
@@ -0,0 +1,52 @@
+report_lines ()
+{
+    local line
+
+    if [[ ${#lines[@]} -gt 0 ]]; then
+	echo "# "
+    fi
+    
+    for line in "${lines[@]}"; do
+	echo "# $line"
+    done
+} >&3
+
+#
+# simple_test input.peg
+# 
+simple_test ()
+{
+    local cmd
+
+    local input=$1
+    local base=$(basename "$input" .peg)
```

Comment:
There should be quotes, in (unlikely, but possible) case of filename with withspace:
```suggestion
    local input="$1"
    local base="$(basename "$input" .peg)"
```